### PR TITLE
OCLOMRS-33: Implements logic for retrieving and searching for concepts

### PR DESCRIPTION
**Pull request by hadijahkyampeire**
# JIRA TICKET NAME:
[OCLOMRS-33: Implements logic for retrieving and searching for concepts](https://issues.openmrs.org/browse/OCLOMRS-33)

# Summary:
This PR does the following:
- Adds UI for viewing all concepts in OCL
- Implements the logic for retrieving all concepts and searching through them
- Adds tests for all the functionality

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -18,6 +19,7 @@ const App = () => (
         <Switch>
           <Route exact path="/" component={Login} />
           <Route exact path="/dashboard" component={Authenticate(SourceSearch)} />
+          <Route exact path="/dashboard/concepts" component={Authenticate(ConceptSerach)} />
```
Comment (emasys) [inline src/App.js]: There seem to be a typo here

File: src/components/dashboard/components/concepts/ConceptTable.jsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const {
+    concept: {
+      id, display_name, concept_class, datatype, source,
+    },
+  } = concept;
+  return (
+    <tr className="concept-table">
+      <td>{id}</td>
+      <td style={{ maxWidth: '25rem' }} className="display-name">
+        {/* <div style={{ maxWidth: '25rem' }} className="display-name"> */}
```
Comment (emasys) [inline src/components/dashboard/components/concepts/ConceptTable.jsx]: do well to delete these comments

File: src/App.js (hunk 1 above)
Comment (hadijahkyampeire) [reply src/App.js]: What typo @emasys

**Comment by mwaz** [comment]
LGTM

**Comment by EleisonC** [comment]
Why is Travis failing

**Comment by hadijahkyampeire** [comment]
Everything is good now

File: src/components/dashboard/components/concepts/ConceptTable.jsx (hunk 2 above)
Comment (hadijahkyampeire) [reply src/components/dashboard/components/concepts/ConceptTable.jsx]: cool

**Pull request merged**